indexing-service/src/main/java/org/apache/druid/indexing/common/task/Task.java

```diff
@@ -313,6 +315,15 @@ default TaskIdentifier getMetadata()
     return new TaskIdentifier(this.getId(), this.getGroupId(), this.getType());
   }
 
+  /**
+   * @return The status of the task. Note: this interface method is unstable at this time.
+   */
+  @Nullable
+  default String getStatus()
```

Comment:
Rather than adding a `getStatus()` method here, I would prefer it if we just cast to `SeekableStreamTask` in `CliPeon`.

The `Task.run()` method runs the task and returns the final status.
We shouldn't have or need access to the status of the task while the task is running.

Moreover, the status being returned here is not the same as the task status.

Reply:
Fixed, let me know if ok now.